drivers/flash/flash_page_layout.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <flash.h>
+
+static int _flash_get_page_info(struct device *dev, off_t offs,
+				   bool use_addr, struct flash_pages_info *info)
+{
+	const struct flash_driver_api *api = dev->driver_api;
+	const struct flash_pages_layout *layout;
+	size_t page_count = 0;
+	off_t group_offs = 0;
+	u32_t num_in_group;
+	off_t end = 0;
+	size_t layout_size;
+
+	api->page_layout(dev, &layout, &layout_size);
```

Comment:
Do we expect page layouts to ever change, seems like we should call page_layout once in an initcall.

Reply:
It seems we should make this requirement in general that the given flash drivers 'init' function call the page_layout() function to setup things.

Reply:
I would keep this up to the implementation because this doesn't break anything. Some external serial flash memories are able to change the layout (on a command).

Reply:
but yes, it is better to initialize layout description in init call - not neccery using page_layout() api.

Reply:
Initialization was moved to init. call for nrf5x.